aws/resources/ec2.go

```diff
@@ -140,722 +130,34 @@ func (ei *EC2Instances) nukeAll(instanceIds []*string) error {
-		securityGroupRules, _ := v.svc.DescribeSecurityGroupRules(
-			&ec2.DescribeSecurityGroupRulesInput{
-				Filters: []*ec2.Filter{
-					{
-						Name:   awsgo.String("group-id"),
-						Values: []*string{securityGroup.GroupId},
-					},
-				},
-			},
-		)
-		for _, securityGroupRule := range securityGroupRules.SecurityGroupRules {
-			msg := fmt.Sprintf("...deleting Security Group Rule %s", awsgo.StringValue(securityGroupRule.SecurityGroupRuleId))
-			spinner.UpdateText(msg)
-			logging.Debug(msg)
-			if *securityGroupRule.IsEgress {
-				_, err := v.svc.RevokeSecurityGroupEgress(
-					&ec2.RevokeSecurityGroupEgressInput{
-						GroupId:              securityGroup.GroupId,
-						SecurityGroupRuleIds: []*string{securityGroupRule.SecurityGroupRuleId},
-					},
-				)
-				if err != nil {
-					return errors.WithStackTrace(err)
-				}
-			} else {
-				_, err := v.svc.RevokeSecurityGroupIngress(
-					&ec2.RevokeSecurityGroupIngressInput{
-						GroupId:              securityGroup.GroupId,
-						SecurityGroupRuleIds: []*string{securityGroupRule.SecurityGroupRuleId},
-					},
-				)
-				if err != nil {
-					return errors.WithStackTrace(err)
-				}
-			}
-		}
-	}
-
-	for _, securityGroup := range securityGroups.SecurityGroups {
-		msg := fmt.Sprintf("...deleting Security Group %s", awsgo.StringValue(securityGroup.GroupId))
-		spinner.UpdateText(msg)
-		logging.Debug(msg)
-		if *securityGroup.GroupName != "default" {
-			_, err := v.svc.DeleteSecurityGroup(
-				&ec2.DeleteSecurityGroupInput{
-					GroupId: securityGroup.GroupId,
-				},
-			)
-			if err != nil {
-				return errors.WithStackTrace(err)
-			}
-		}
-	}
-	return nil
-}
-
-func (v Vpc) nukeEndpoints(spinner *pterm.SpinnerPrinter) error {
-	endpoints, _ := v.svc.DescribeVpcEndpoints(
-		&ec2.DescribeVpcEndpointsInput{
-			Filters: []*ec2.Filter{
-				{
-					Name:   awsgo.String("vpc-id"),
-					Values: []*string{awsgo.String(v.VpcId)},
-				},
-			},
-		},
-	)
-
-	var endpointIds []*string
-
-	for _, endpoint := range endpoints.VpcEndpoints {
-		endpointIds = append(endpointIds, endpoint.VpcEndpointId)
-		msg := fmt.Sprintf("...deleting VPC endpoint %s", awsgo.StringValue(endpoint.VpcEndpointId))
-		spinner.UpdateText(msg)
-		logging.Debugf(msg)
-	}
-
-	if len(endpointIds) == 0 {
-		msg := "...no endpoints found"
-		spinner.UpdateText(msg)
-		logging.Debug(msg)
-		return nil
-	}
-
-	_, err := v.svc.DeleteVpcEndpoints(&ec2.DeleteVpcEndpointsInput{
-		VpcEndpointIds: endpointIds,
-	})
-	if err != nil {
-		return errors.WithStackTrace(err)
-	}
-
-	if err := waitForVPCEndpointsToBeDeleted(v); err != nil {
-		return errors.WithStackTrace(err)
-	}
-
-	return nil
-}
-
-func (v Vpc) nukeEgressOnlyGateways(spinner *pterm.SpinnerPrinter) error {
-	allEgressGateways := []*string{}
-	msg := "Finding Egress Only Internet Gateways to Nuke"
-	spinner.UpdateText(msg)
-	logging.Debug(msg)
-	err := v.svc.DescribeEgressOnlyInternetGatewaysPages(
```

Comment:
I was not sure how to implement the `paginator` for this if we don't have access to `ei.Client`

Reply:
by looking at the code you should use the client under `svc` on the `VPC type` e.q.

```go
	v.svc.ClientAPI_Call
```

but first you also need to update the Client on the **VPC** type with **new sdkV2** check line `180`